fetch.bs

```diff
@@ -6892,30 +6879,58 @@ returns failure or a <a for=/>MIME type</a>.
 <a>this</a>'s <a for=Body>body</a> is non-null and <a>this</a>'s <a for=Body>body</a>'s
 <a for=body>stream</a> is <a for=ReadableStream>disturbed</a>; otherwise false.
 
-<p>The <dfn id=concept-body-package-data for=Body>package data</dfn> algorithm, given
-<var>bytes</var>, <var>type</var>, and a <var>mimeType</var>, switches on <var>type</var>, and runs
-the associated steps:
+<p>The <dfn id=concept-body-consume-body for=Body>consume body</dfn> algorithm, given a {{Body}}
+<var>object</var> and <var>convertBytesToJSValue</var>, which is an algorithm that accepts a
+<a for=/>byte sequence</a> and returns a JavaScript value or throws an exception, runs these steps:
 
-<dl class=switch>
- <dt><i>ArrayBuffer</i>
- <dd>
-  <p>Return a new {{ArrayBuffer}} whose contents are <var>bytes</var>.
+<ol>
+ <li><p>If <var>object</var> is <a for=Body>unusable</a>, then return <a>a promise rejected with</a>
+ a {{TypeError}}.
 
-  <p class=note>Allocating an {{ArrayBuffer}} can throw a {{RangeError}}.
+ <li><p>Let <var>promise</var> be <a>a new promise</a>.
 
- <dt><i>Blob</i>
- <dd><p>Return a {{Blob}} whose contents are <var>bytes</var> and {{Blob/type}} attribute is
- <var>mimeType</var>.
+ <li>Let <var>successSteps</var> given a <a for=/>byte sequence</a> <var>data</var> be to
+ <a for=/>resolve</a> <var>promise</var> with the result of running <var>convertBytesToJSValue</var>
+ with <var>data</var>. If that threw an exception, <a>reject</a> <var>promise</var> with that
+ exception.
 
- <dt><i>FormData</i>
- <dd>
-  <p>If <var>mimeType</var>'s <a for="MIME type">essence</a> is "<code>multipart/form-data</code>",
-  then:
+ <li><p>If <var>object</var>'s <a for=Body>body</a> is null, then run <var>successSteps</var>
+ with an empty <a for=/>byte sequence</a>, and return.
+
+ <li>Let <var>errorSteps</var> given <var>error</var> be to <a>reject</a> <var>promise</var> with
+ <var>error</var>.
```

Comment:
If you define these before _successSteps_ you can invoke them as part of defining _successSteps_.

Reply:
Done